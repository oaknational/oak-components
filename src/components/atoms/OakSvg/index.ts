export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
