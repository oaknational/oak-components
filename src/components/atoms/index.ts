export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
