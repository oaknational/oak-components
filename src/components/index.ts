export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
