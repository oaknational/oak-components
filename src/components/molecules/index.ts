export * from "./OakFieldError";
export * from "./OakLoadingSpinner";
export * from "./OakPrimaryButton";
export * from "./OakPrimaryInvertedButton";
export * from "./OakSecondaryButton";
export * from "./OakTertiaryButton";
export * from "./OakRadioButton";
export * from "./OakRadioGroup";
export * from "./OakCheckBox";
export * from "./OakTextInput";
export * from "./OakRoundIcon";
export * from "./OakTooltip";
export * from "./OakBackLink";
export * from "./OakBulletList";
export * from "./OakHandDrawnCard";
export * from "./OakHandDrawnCardWithIcon";
export * from "./OakHandDrawnHR";
export * from "./OakPromoTag";
export * from "./OakLessonInfoCard";
export * from "./OakCollapsibleContent";
export * from "./OakLink";
export * from "./OakSecondaryLink";
export * from "./OakDragAndDropInstructions";
export * from "./OakDraggable";
export * from "./OakDroppable";
export * from "./OakDraggableFeedback";
export * from "./OakOutlineAccordion";
export * from "./OakAccordion";
export * from "./OakModal";
export * from "./OakModalCenter";
export * from "./OakCardWithHandDrawnBorder";
export * from "./OakSecondaryButtonAsRadio";
export * from "./OakButtonAsRadioGroup";
