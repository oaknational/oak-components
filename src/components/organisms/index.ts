export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
