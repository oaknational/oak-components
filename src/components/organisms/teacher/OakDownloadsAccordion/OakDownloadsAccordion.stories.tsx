import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";

import {
  OakDownloadsAccordion,
  OakDownloadsAccordionProps,
} from "./OakDownloadsAccordion";

import { OakFlex } from "@/components/atoms";
import { OakDownloadCard } from "@/components/molecules/OakDownloadCard";
import { colorArgTypes } from "@/storybook-helpers/colorStyleHelpers";
import { borderArgTypes } from "@/storybook-helpers/borderStyleHelpers";

const meta: Meta<typeof OakDownloadsAccordion> = {
  title: "Components/Organisms/teacher/OakDownloadsAccordion",
  component: OakDownloadsAccordion,
  tags: ["autodocs"],
  parameters: {
    controls: {
      include: ["initialOpen"],
    },
  },
  argTypes: {
    children: {
      control: {
        type: "text",
      },
    },
    initialOpen: {
      control: {
        type: "boolean",
      },
    },
    ...colorArgTypes,
    ...borderArgTypes,
  },
};

export default meta;

type Story = StoryObj<typeof OakDownloadsAccordion>;

const WithState = (args: OakDownloadsAccordionProps) => {
  const [checked, setChecked] = useState(false);
  const handleChecked = () => {
    setChecked((prev) => !prev);
  };
  return (
    <OakDownloadsAccordion
      {...args}
      selectAllChecked={checked}
      handleToggleSelectAll={handleChecked}
    />
  );
};

export const Default: Story = {
  args: {
    id: "oak-downloads-accordion",
    downloadsText: "Slides, quizzes, worksheet, additional materials",
    children: (
      <OakFlex $flexDirection="column" $gap="space-between-s">
        <OakDownloadCard
          id="lesson-plan"
          value="lesson-plan"
          iconName="book-steps"
          titleSlot="Lesson plan"
          formatSlot="PDF"
          fileSizeSlot="245KB"
        />
        <OakDownloadCard
          id="presentation"
          value="presentation"
          iconName="slide-deck"
          titleSlot="Lesson presentation"
          formatSlot="PPTX"
          fileSizeSlot="1.2MB"
        />
        <OakDownloadCard
          id="worksheet"
          value="worksheet"
          iconName="worksheet"
          titleSlot="Student worksheet"
          formatSlot="PDF"
          fileSizeSlot="180KB"
        />
      </OakFlex>
    ),
    initialOpen: false,
    $width: "100%",
  },
  render: (args) => <WithState {...args} />,
};
