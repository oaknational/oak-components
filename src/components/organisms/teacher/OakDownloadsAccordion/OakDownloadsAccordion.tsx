import React, { ReactNode } from "react";
import styled from "styled-components";

import {
  OakBox,
  OakFlex,
  OakIcon,
  OakBoxProps,
  oakBoxCss,
  OakHeading,
  OakP,
} from "@/components/atoms";
import {
  InternalAccordionButton,
  InternalAccordionContent,
} from "@/components/atoms/InternalAccordion";
import useAccordionContext from "@/components/atoms/InternalAccordion/useAccordionContext";
import InternalAccordionProvider from "@/components/atoms/InternalAccordion/InternalAccordionProvider";
import { FlexStyleProps, flexStyle } from "@/styles/utils/flexStyle";
import { ColorStyleProps } from "@/styles/utils/colorStyle";
import { parseDropShadow } from "@/styles/helpers/parseDropShadow";
import { parseColorFilter } from "@/styles/helpers/parseColorFilter";
import { OakCheckBox } from "@/components/molecules";

export type OakDownloadsAccordionProps = {
  /**
   * Text to render in the subheading
   */
  downloadsText: string;
  /**
   * Event handler for select all checkbox
   */
  handleToggleSelectAll: () => void;
  /**
   * State of select all checkbox
   */
  selectAllChecked: boolean;
  /**
   * The content of the accordion
   */
  children: ReactNode;
  /**
   * The id of the accordion
   */
  id: string;
  /**
   * Whether the accordion starts in the open state, defaults to false
   */
  initialOpen?: boolean;
} & FlexStyleProps &
  OakBoxProps &
  ColorStyleProps;

const StyledAccordionButton = styled(InternalAccordionButton)<FlexStyleProps>`
  ${flexStyle}
  ${oakBoxCss}
  min-width: 0;
  &:hover {
    .chevron-icon {
      filter: ${parseColorFilter("grey60")};
    }
  }
  &:focus-visible {
    .focus-outline {
      box-shadow: ${parseDropShadow("drop-shadow-centered-lemon")},
        ${parseDropShadow("drop-shadow-centered-grey")};
    }
  }
`;

/**
 * Internal accordion component that handles the accordion logic
 */
const Accordion = ({
  children,
  id,
  selectAllChecked,
  downloadsText,
  handleToggleSelectAll,
  ...styleProps
}: OakDownloadsAccordionProps) => {
  const { isOpen } = useAccordionContext();

  return (
    <OakFlex
      $bt={"border-solid-s"}
      $bb={isOpen ? "border-solid-none" : "border-solid-s"}
      $borderColor={"grey40"}
      $pv={"inner-padding-s"}
      $flexDirection={"column"}
      $width={"100%"}
      {...styleProps}
    >
      <OakFlex $alignItems={"center"} $width={"100%"}>
        <OakFlex
          $alignItems={"center"}
          $width={"100%"}
          $gap={"space-between-s"}
        >
          <OakBox
            id="select-all-wrapper"
            $pa={"inner-padding-xs"}
            onClick={handleToggleSelectAll}
          >
            <OakCheckBox
              onChange={() => undefined}
              checked={selectAllChecked}
              id="select-all"
              name="select-all"
              value={""}
              aria-labelledby="downloads-accordion-heading"
            />
          </OakBox>
          <StyledAccordionButton
            $justifyContent={"space-between"}
            $flexGrow={1}
            id={id}
            $alignItems={"center"}
            $mr={"space-between-xs"}
          >
            <OakFlex
              $justifyContent={"center"}
              $alignItems={"center"}
              $gap={"space-between-xs"}
            >
              <OakFlex
                $flexDirection={"column"}
                $justifyContent={"center"}
                $alignItems={"start"}
              >
                <OakHeading
                  $textAlign={"left"}
                  id="downloads-accordion-heading"
                  $color={"text-primary"}
                  $font={"heading-6"}
                  $mr={"space-between-s"}
                  tag="h2"
                >
                  {selectAllChecked
                    ? "All resources selected"
                    : "Select all resources"}
                </OakHeading>
                <OakBox $pr={"inner-padding-s"}>
                  <OakP $textAlign={"left"}>{downloadsText}</OakP>
                </OakBox>
              </OakFlex>
            </OakFlex>
            <OakBox
              className="focus-outline"
              $borderRadius={"border-radius-s"}
              $pa={"inner-padding-xs"}
            >
              <OakIcon
                className="chevron-icon"
                iconName="chevron-down"
                $width="all-spacing-7"
                $height="all-spacing-7"
                alt="An arrow to indicate whether the item is open or closed"
                style={{
                  transform: isOpen ? "rotate(180deg)" : "none",
                  transition: "all 0.3s ease 0s",
                }}
              />
            </OakBox>
          </StyledAccordionButton>
        </OakFlex>
      </OakFlex>
      {/* Accordion content */}
      <OakBox $position={"relative"} $overflow={"auto"}>
        <InternalAccordionContent aria-labelledby={id}>
          {children}
        </InternalAccordionContent>
      </OakBox>
    </OakFlex>
  );
};

/**
 * OakDownloadsAccordion
 *
 */
export const OakDownloadsAccordion = ({
  initialOpen = false,
  ...props
}: OakDownloadsAccordionProps) => {
  return (
    <InternalAccordionProvider isInitialOpen={initialOpen}>
      <Accordion {...props} />
    </InternalAccordionProvider>
  );
};
