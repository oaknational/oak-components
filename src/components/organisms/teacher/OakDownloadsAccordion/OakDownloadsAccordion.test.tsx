import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";
import { act, fireEvent } from "@testing-library/react";

import { OakDownloadsAccordion } from "./OakDownloadsAccordion";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { OakThemeProvider, OakFlex } from "@/components/atoms";
import { oakDefaultTheme } from "@/styles";
import { OakDownloadCard } from "@/components/molecules/OakDownloadCard";

const mockHandleToggleSelectAll = jest.fn();

const defaultProps = {
  id: "test-downloads-accordion",
  downloadsText: "Slides, quizzes, worksheet, additional materials",
  handleToggleSelectAll: mockHandleToggleSelectAll,
  selectAllChecked: false,
  children: (
    <OakFlex $flexDirection="column" $gap="spacing-16">
      <OakDownloadCard
        id="lesson-plan"
        value="lesson-plan"
        iconName="book-steps"
        titleSlot="Lesson plan"
        formatSlot="PDF"
        fileSizeSlot="245KB"
      />
      <OakDownloadCard
        id="presentation"
        value="presentation"
        iconName="slide-deck"
        titleSlot="Lesson presentation"
        formatSlot="PPTX"
        fileSizeSlot="1.2MB"
      />
    </OakFlex>
  ),
};

describe("OakDownloadsAccordion", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("matches snapshot when closed", () => {
    const tree = create(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakDownloadsAccordion {...defaultProps} />
      </OakThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays correct heading when selectAllChecked is false", () => {
    const { getByText } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} selectAllChecked={false} />,
    );

    expect(getByText("Select all resources")).toBeInTheDocument();
  });

  it("displays correct heading when selectAllChecked is true", () => {
    const { getByText } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} selectAllChecked={true} />,
    );

    expect(getByText("All resources selected")).toBeInTheDocument();
  });

  it("displays the downloads text", () => {
    const { getByText } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} />,
    );

    expect(
      getByText("Slides, quizzes, worksheet, additional materials"),
    ).toBeInTheDocument();
  });

  it("starts closed by default", () => {
    const { queryByRole } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} />,
    );

    expect(queryByRole("region")).not.toBeInTheDocument();
  });

  it("starts open when initialOpen is true", () => {
    const { getByRole, getByText } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} initialOpen={true} />,
    );

    expect(getByRole("region")).toBeInTheDocument();
    expect(getByText("Lesson plan")).toBeInTheDocument();
    expect(getByText("Lesson presentation")).toBeInTheDocument();
  });

  it("toggles open and closed when accordion button is clicked", () => {
    const { queryByRole, getByRole, getByText } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} initialOpen={false} />,
    );

    expect(queryByRole("region")).not.toBeInTheDocument();

    act(() => {
      fireEvent.click(getByRole("button"));
    });

    expect(getByRole("region")).toBeInTheDocument();
    expect(getByText("Lesson plan")).toBeInTheDocument();

    act(() => {
      fireEvent.click(getByRole("button"));
    });

    expect(queryByRole("region")).not.toBeInTheDocument();
  });

  it("calls handleToggleSelectAll when select all checkbox clicked", () => {
    const { getByRole } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} />,
    );

    act(() => {
      fireEvent.click(getByRole("checkbox"));
    });

    expect(mockHandleToggleSelectAll).toHaveBeenCalledTimes(1);
  });

  it("calls handleToggleSelectAll when select all wrapper is clicked", () => {
    const { container } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} />,
    );

    const selectAllWrapper = container.querySelector("#select-all-wrapper");

    if (selectAllWrapper) {
      act(() => {
        fireEvent.click(selectAllWrapper);
      });

      expect(mockHandleToggleSelectAll).toHaveBeenCalledTimes(1);
    } else {
      // If the wrapper doesn't exist, skip this test or fail appropriately
      expect(selectAllWrapper).toBeInTheDocument();
    }
  });

  it("reflects the checked state of the select all checkbox", () => {
    const { getByRole, rerender } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} selectAllChecked={false} />,
    );

    const checkbox = getByRole("checkbox");
    expect(checkbox).not.toBeChecked();

    rerender(
      <OakDownloadsAccordion {...defaultProps} selectAllChecked={true} />,
    );

    expect(checkbox).toBeChecked();
  });

  it("has correct accessibility attributes", () => {
    const { getByRole } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} />,
    );

    const button = getByRole("button");
    const checkbox = getByRole("checkbox");

    expect(button).toHaveAttribute("id", "test-downloads-accordion");
    expect(checkbox).toHaveAttribute("id", "select-all");
    expect(checkbox).toHaveAttribute("name", "select-all");
    expect(checkbox).toHaveAttribute(
      "aria-labelledby",
      "downloads-accordion-heading",
    );
  });

  it("displays chevron icon that rotates when accordion opens", () => {
    const { getByRole } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} initialOpen={false} />,
    );

    const chevronIcon = document.querySelector(".chevron-icon");
    expect(chevronIcon).toBeInTheDocument();
    expect(chevronIcon).toHaveStyle("transform: none");

    act(() => {
      fireEvent.click(getByRole("button"));
    });

    expect(chevronIcon).toHaveStyle("transform: rotate(180deg)");
  });

  it("renders children content when open", () => {
    const { getByText } = renderWithTheme(
      <OakDownloadsAccordion {...defaultProps} initialOpen={true} />,
    );

    expect(getByText("Lesson plan")).toBeInTheDocument();
    expect(getByText("Lesson presentation")).toBeInTheDocument();
    expect(getByText("PDF")).toBeInTheDocument();
    expect(getByText("PPTX")).toBeInTheDocument();
  });
});
