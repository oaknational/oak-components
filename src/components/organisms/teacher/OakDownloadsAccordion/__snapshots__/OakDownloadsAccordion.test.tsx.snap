// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakDownloadsAccordion matches snapshot when closed 1`] = `
.c0 {
  width: 100%;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  border-top: 0.063rem solid;
  border-bottom: 0.063rem solid;
  border-color: #cacaca;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c2 {
  width: 100%;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c5 {
  padding: 0.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c5:hover {
  cursor: pointer;
}

.c8 {
  position: relative;
  width: 1.5rem;
  height: 1.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c14 {
  position: absolute;
  top: 0rem;
  left: 0rem;
  width: 1.5rem;
  height: 1.5rem;
  padding: 0rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c16 {
  position: relative;
  width: 100%;
  min-width: 100%;
  height: 100%;
  min-height: 100%;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c18 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c18:hover {
  cursor: pointer;
}

.c22 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c26 {
  padding-right: 0.75rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c28 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c29 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c31 {
  position: relative;
  overflow: auto;
  font-family: --var(google-font),Lexend,sans-serif;
  overflow: auto;
}

.c32 {
  padding-top: 1.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c34 {
  overflow: hidden;
  color: #222222;
  background: #ffffff;
  border: 0.125rem solid;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
  overflow: hidden;
}

.c39 {
  padding: 0.75rem;
  background: #ffe555;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c41 {
  position: relative;
  width: 3rem;
  min-width: 3rem;
  height: 3rem;
  min-height: 3rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c42 {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c44 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c45 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c46 {
  padding-right: 1rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c17 {
  -webkit-filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  object-fit: contain;
}

.c30 {
  object-fit: contain;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.75rem;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c38 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c40 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.c43 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  gap: 0.25rem;
}

.c25 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
  text-align: left;
  margin-right: 1rem;
  color: #222222;
}

.c6 {
  font-family: --var(google-font),Lexend,sans-serif;
  color: #222222;
}

.c27 {
  font-family: --var(google-font),Lexend,sans-serif;
  text-align: left;
}

.c20 {
  font: inherit;
  color: inherit;
  border: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  padding: 0;
}

.c10 {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  margin: 0;
  outline: none;
  border: 0.125rem solid;
  border-color: #222222;
  border-radius: 0.125rem;
  width: 1.5rem;
  height: 1.5rem;
  border-color: #808080;
}

.c10:checked {
  background: #222222;
  border-color: #222222;
}

.c10:disabled {
  pointer-events: none;
}

.c12:focus-visible {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 {
  pointer-events: none;
  opacity: 0;
}

input:checked + .c13 {
  opacity: 1;
}

.c37 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c36 {
  cursor: pointer;
}

.c36:has(input:focus-within):focus-visible {
  box-shadow: 0px 0px 0px 2px #ffe555, 0px 0px 0px 5px #575757;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-right: 0.75rem;
  font-family: --var(google-font),Lexend,sans-serif;
  min-width: 0;
}

.c21:hover .chevron-icon {
  -webkit-filter: invert(34%) sepia(0%) saturate(698%) hue-rotate(158deg) brightness(95%) contrast(89%);
  filter: invert(34%) sepia(0%) saturate(698%) hue-rotate(158deg) brightness(95%) contrast(89%);
}

.c21:focus-visible .focus-outline {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

@media (hover:hover) {
  .c10:hover:not(:disabled) {
    border-color: #222222;
  }
}

@media (hover:hover) {
  .c11:hover:not(.c9:checked):not(.c9:disabled)::after {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    width: 60%;
    height: 60%;
    -webkit-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    border-radius: 0.125rem;
    background: #222222;
  }
}

@media (hover:hover) {
  .c36:hover:not(:disabled,:active) {
    background: #f2f2f2;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c2 c4"
      >
        <div
          class="c5"
          id="select-all-wrapper"
        >
          <label
            class="c6 c7"
            for="select-all"
          >
            <div
              class="c8"
            >
              <input
                aria-labelledby="downloads-accordion-heading"
                class="c9 c10 c11 c12"
                id="select-all"
                name="select-all"
                type="checkbox"
                value=""
              />
              <div
                class="c13 c14 c15"
              >
                <div
                  class="c16"
                >
                  <img
                    alt=""
                    class="c17"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699954310/icons/efd3esaor6zqk7seh6kt.svg"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                  />
                </div>
              </div>
            </div>
          </label>
        </div>
        <button
          aria-expanded="false"
          class="c18 c19 c20  c21"
          id="test-downloads-accordion"
          type="button"
        >
          <div
            class="c22 c23"
          >
            <div
              class="c22 c24"
            >
              <h2
                class="c25"
                id="downloads-accordion-heading"
              >
                Select all resources
              </h2>
              <div
                class="c26"
              >
                <p
                  class="c27"
                >
                  Slides, quizzes, worksheet, additional materials
                </p>
              </div>
            </div>
          </div>
          <div
            class="c28 focus-outline"
          >
            <div
              class="c29 chevron-icon"
              style="transform: none; transition: all 0.3s ease 0s;"
            >
              <img
                alt="An arrow to indicate whether the item is open or closed"
                class="c30"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699953557/icons/botfld6brychmttwtv6u.svg"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      class="c31"
    >
      <div
        aria-labelledby="test-downloads-accordion"
        class="c32"
        hidden=""
        role="region"
      >
        <div
          class="c22 c33"
        >
          <div
            class="c34 c35 c36"
          >
            <label
              class="c37"
            >
              <div
                class="c22 c38"
              >
                <div
                  class="c39 c40"
                >
                  <div
                    class="c41"
                  >
                    <img
                      alt=""
                      class="c30"
                      data-nimg="fill"
                      decoding="async"
                      loading="lazy"
                      src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1749034739/icons/book-steps_qwku6h.svg"
                      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                    />
                  </div>
                </div>
                <div
                  class="c42 c43"
                >
                  <div
                    class="c44"
                  >
                    Lesson plan
                  </div>
                  <div
                    class="c45"
                  >
                    245KB
                  </div>
                  <div
                    class="c45"
                  >
                    PDF
                  </div>
                </div>
              </div>
              <div
                class="c22 c35"
              >
                <div
                  class="c46 c3"
                >
                  <div
                    class="c8"
                  >
                    <input
                      class="c9 c10 c11 c12"
                      id="lesson-plan"
                      name="default"
                      type="checkbox"
                      value="lesson-plan"
                    />
                    <div
                      class="c13 c14 c15"
                    >
                      <div
                        class="c16"
                      >
                        <img
                          alt=""
                          class="c17"
                          data-nimg="fill"
                          decoding="async"
                          loading="lazy"
                          src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699954310/icons/efd3esaor6zqk7seh6kt.svg"
                          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </label>
          </div>
          <div
            class="c34 c35 c36"
          >
            <label
              class="c37"
            >
              <div
                class="c22 c38"
              >
                <div
                  class="c39 c40"
                >
                  <div
                    class="c41"
                  >
                    <img
                      alt=""
                      class="c30"
                      data-nimg="fill"
                      decoding="async"
                      loading="lazy"
                      src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699954241/icons/sjjy5f3g4eciwcuxxr33.svg"
                      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                    />
                  </div>
                </div>
                <div
                  class="c42 c43"
                >
                  <div
                    class="c44"
                  >
                    Lesson presentation
                  </div>
                  <div
                    class="c45"
                  >
                    1.2MB
                  </div>
                  <div
                    class="c45"
                  >
                    PPTX
                  </div>
                </div>
              </div>
              <div
                class="c22 c35"
              >
                <div
                  class="c46 c3"
                >
                  <div
                    class="c8"
                  >
                    <input
                      class="c9 c10 c11 c12"
                      id="presentation"
                      name="default"
                      type="checkbox"
                      value="presentation"
                    />
                    <div
                      class="c13 c14 c15"
                    >
                      <div
                        class="c16"
                      >
                        <img
                          alt=""
                          class="c17"
                          data-nimg="fill"
                          decoding="async"
                          loading="lazy"
                          src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699954310/icons/efd3esaor6zqk7seh6kt.svg"
                          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
